src/test/scala/com/mozilla/telemetry/streaming/TestAggregator.scala

```diff
@@ -100,4 +100,55 @@ class TestAggregator extends FlatSpec with Matchers{
     results("window_start").asInstanceOf[Timestamp].getTime should be <= (TestUtils.testTimestampMillis)
     results("window_end").asInstanceOf[Timestamp].getTime should be >= (TestUtils.testTimestampMillis)
   }
-}
+
+  "The aggregator" should "handle new style experiments" in {
+    val spark = SparkSession.builder()
+      .appName("Error Aggregates")
+      .master("local[1]")
+      .getOrCreate()
+    import spark.implicits._
+    val crashMessage = TestUtils.generateCrashMessages(
+      k,
+      Some(Map(
+        "environment.addons" ->
+          """
+            |{
+            | "activeAddons": {"my-addon": {"isSystem": true}},
+            | "theme": {"id": "firefox-compact-dark@mozilla.org"}
+            |}""".stripMargin,
+        "environment.experiments" ->
+          """
+            |{
+            |  "new-experiment-1": {"branch": "control"},
+            |  "new-experiment-2": {"branch": "chaos"}
+            |}""".stripMargin
+      )))
+    val mainMessage = TestUtils.generateMainMessages(
+      k,
+      Some(Map(
+        "environment.addons" ->
+          """
+            |{
+            | "activeAddons": {"my-addon": {"isSystem": true}},
+            | "theme": {"id": "firefox-compact-dark@mozilla.org"}
+            |}""".stripMargin,
+        "environment.experiments" ->
+          """
+            |{
+            |  "new-experiment-1": {"branch": "control"},
+            |  "new-experiment-2": {"branch": "chaos"}
+            |}""".stripMargin
+      )))
+    val messages = (crashMessage ++ mainMessage).map(_.toByteArray).seq
+    val df = ErrorAggregator.aggregate(spark.sqlContext.createDataset(messages).toDF, raiseOnError = true, online = false)
+    df.count() should be (1)
+    val inspectedFields = List(
+      "experiment_id",
+      "experiment_branch"
+    )
+    val row = df.select(inspectedFields(0), inspectedFields.drop(1):_*).first()
+    val results = 0.to(inspectedFields.length-1).map(x => (inspectedFields(x), row(x))).toMap
```

Comment:
This can be replaced with `row.toSeq.zip(inspectedFields).toMap`, which is more readable.